plumbing/transport/internal/common/common.go

```diff
@@ -46,6 +46,8 @@ type Commander interface {
 type Command interface {
 	// SetAuth sets the authentication method.
 	SetAuth(transport.AuthMethod) error
+	// Perform the actual network connection.
+	Connect() error
```

Comment:
SetAuth is actually a leftover from a previous implementation that should have been gone. Auth can only be set before connection and we don't support reusing commands or connections at all.

So, instead of adding a new Connect function, I think it would be better to just remove SetAuth from all interfaces (Command, Session) and add transport.AuthMethod as a second parameter to NewUploadPackSession and NewReceivePackSession.

That would keep these interfaces simpler and avoid invalid sequences of function calls.